libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,207 @@
+    {
+#if DEBUG
+        public
+#else
+        private
+#endif
+        string userAgent;
+
+        private IBot bot;
+
+        private IServiceProvider services;
+
+        private IList<IMiddleware> middlewareSet;
+
+        private Func<ITurnContext, Exception, Task> onTurnError;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingRequestHandler"/> class.
+        /// The StreamingRequestHandler serves as a translation layer between the transport layer and bot adapter.
+        /// It receives ReceiveRequests from the transport and provides them to the bot adapter in a form
+        /// it is able to build activities out of, which are then handed to the bot itself to processed.
+        /// Throws <see cref="ArgumentNullException"/> if arguments are null.
+        /// </summary>
+        /// <param name="onTurnError">Optional function to perform on turn errors.</param>
+        /// <param name="bot">The <see cref="IBot"/> to be used for all requests to this handler.</param>
+        /// <param name="middlewareSet">An optional set of middleware to register with the bot.</param>
+        public StreamingRequestHandler(Func<ITurnContext, Exception, Task> onTurnError, IBot bot, IList<IMiddleware> middlewareSet = null)
+        {
+            this.bot = bot ?? throw new ArgumentNullException(nameof(bot));
+            this.middlewareSet = middlewareSet;
+            userAgent = GetUserAgent();
+            this.onTurnError = onTurnError;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingRequestHandler"/> class.
+        /// An overload for use with dependency injection via ServiceProvider, as shown
+        /// in DotNet Core Bot Builder samples.
+        /// Throws <see cref="ArgumentNullException"/> if arguments are null.
+        /// </summary>
+        /// <param name="onTurnError">Optional function to perform on turn errors.</param>
+        /// <param name="serviceProvider">The service collection containing the registered IBot type.</param>
+        /// <param name="middlewareSet">An optional set of middleware to register with the bot.</param>
+        public StreamingRequestHandler(Func<ITurnContext, Exception, Task> onTurnError, IServiceProvider serviceProvider, IList<IMiddleware> middlewareSet = null)
+        {
+            services = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+            this.middlewareSet = middlewareSet;
+            userAgent = GetUserAgent();
+            this.onTurnError = onTurnError;
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="IStreamingTransportServer"/> this handler listens for incoming messages on.
+        /// </summary>
+        /// <value>
+        /// The <see cref="IStreamingTransportServer"/> this handler listens for incoming messages on.
+        /// </value>
+        public IStreamingTransportServer Server { get; set; }
+
```

Comment:
I don't get why this isn't a constructor parameter? It seems really weird (concurrency wise) that this can be changed while requests are in-flight.